lib/Parser/TableParser.php

```diff
@@ -41,53 +49,87 @@ public function parseTableLine(string $line) : ?array
 
         if ($chars[0] === self::PRETTY_TABLE_JOINT && $chars[1] === self::PRETTY_TABLE_LETTER) {
             $pretty = true;
-            $chars  = [self::PRETTY_TABLE_LETTER, self::PRETTY_TABLE_JOINT];
+            // reverse the chars: - is the line char, + is the space char
+            $chars = [self::PRETTY_TABLE_LETTER, self::PRETTY_TABLE_JOINT];
         } elseif ($chars[0] === self::PRETTY_TABLE_JOINT && $chars[1] === self::PRETTY_TABLE_HEADER) {
             $pretty = true;
             $header = true;
-            $chars  = [self::PRETTY_TABLE_HEADER, self::PRETTY_TABLE_JOINT];
+            // reverse the chars: = is the line char, + is the space char
+            $chars = [self::PRETTY_TABLE_HEADER, self::PRETTY_TABLE_JOINT];
         } else {
-            if (! ($chars[0] === self::TABLE_LETTER && $chars[1] === ' ')) {
+            // either a simple table or not a separator line
+
+            // if line char is not "=" or "-", not a separator line
+            if (! in_array($chars[0], [self::SIMPLE_TABLE_LETTER, self::SIMPLE_TABLE_LETTER_ALT], true)) {
+                return null;
+            }
+
+            // if space char is not a space, not a separator line
+            if ($chars[1] !== ' ') {
                 return null;
             }
         }
 
-        $parts     = [];
-        $separator = false;
+        $parts = [];
+        /** @var int|null $currentPartStart */
+        $currentPartStart = null;
 
+        $i = 0;
```

Comment:
Added this because phpstan thought that `$i` might not exist down on line 105 (after the for):

```php
$parts[] = [$currentPartStart, $i];
```

My impression was that `$i` would be initialized and set to 0, even if the `for` loop never looped: it's needed to make the comparison to see if the first loop is needed. So, I initialized it, unnecessarily (which scrutinizer is not happy about).